src/leapfrogai_api/routers/openai/requests/run_create_params_request_base.py

```diff
@@ -0,0 +1,376 @@
+from leapfrogai_api.data.crud_message import CRUDMessage
+from leapfrogai_api.routers.openai.assistants import retrieve_assistant
+from leapfrogai_api.routers.openai.chat import chat_complete, chat_complete_stream_raw
+from leapfrogai_api.backend.converters import (
+    from_assistant_stream_event_to_str,
+    from_text_to_message,
+    from_chat_completion_choice_to_thread_message_delta,
+)
+from leapfrogai_api.routers.openai.requests.create_message_request import (
+    CreateMessageRequest,
+)
+from leapfrogai_api.routers.supabase_session import Session
+from leapfrogai_api.utils import get_model_config
+from leapfrogai_sdk.chat.chat_pb2 import (
+    ChatCompletionResponse as ProtobufChatCompletionResponse,
+)
+
+
+class RunCreateParamsRequestBase(BaseModel):
+    assistant_id: str = Field(default="", examples=["123ab"])
+    instructions: str = Field(default="", examples=["You are a helpful AI assistant."])
+    max_completion_tokens: int | None = Field(default=None, examples=[4096])
+    max_prompt_tokens: int | None = Field(default=None, examples=[32768])
+    metadata: dict | None = Field(default={}, examples=[{}])
+    model: str | None = Field(default=None, examples=["llama-cpp-python"])
+    response_format: AssistantResponseFormatOptionParam | None = Field(
+        default=None, examples=["auto"]
+    )
+    temperature: float | None = Field(default=None, examples=[1.0])
+    tool_choice: AssistantToolChoiceOptionParam | None = Field(
+        default=None, examples=["auto"]
+    )
+    tools: list[AssistantToolParam] = Field(
+        default=[], examples=[[FileSearchToolParam(type="file_search")]]
+    )
+    top_p: float | None = Field(default=None, examples=[1.0])
+    truncation_strategy: TruncationStrategy | None = Field(
+        default=None, examples=[TruncationStrategy(type="auto", last_messages=None)]
+    )
+    parallel_tool_calls: bool | None = Field(default=False, examples=[False])
+
+    @staticmethod
+    def get_initial_messages_base(run: Run) -> list[str]:
+        return [
+            from_assistant_stream_event_to_str(
+                ThreadRunCreated(data=run, event="thread.run.created")
+            ),
+            from_assistant_stream_event_to_str(
+                ThreadRunQueued(data=run, event="thread.run.queued")
+            ),
+            from_assistant_stream_event_to_str(
+                ThreadRunInProgress(data=run, event="thread.run.in_progress")
+            ),
+        ]
+
+    @staticmethod
+    def get_ending_messages_base(run: Run) -> list[str]:
+        return [
+            from_assistant_stream_event_to_str(
+                ThreadRunCompleted(data=run, event="thread.run.completed")
+            )
+        ]
+
+    async def update_with_assistant_data(self, session: Session) -> Assistant:
+        assistant: Assistant | None = await retrieve_assistant(
+            session=session, assistant_id=self.assistant_id
+        )
```

Comment:
```suggestion
        crud_assistant = CRUDAssistant(session)
        assistant: Assistant | None = await crud_assistant.get(filters=FilterAssistant(id=self.assistant_id))
```
You will also need to add this to top:
`from leapfrogai_api.data.crud_assistant import CRUDAssistant, FilterAssistant`

Reply:
This has been resolved